tests/unit/conftest.py

```diff
@@ -30,20 +30,3 @@ def juju_has_secrets(mocker: MockerFixture):
             JujuVersion, "has_secrets", new_callable=PropertyMock
         ).return_value = True
         return True
-
-
-@pytest.fixture
-def only_with_juju_secrets(juju_has_secrets):
-    """Pretty way to skip Juju 3 tests."""
-    if not juju_has_secrets:
-        pytest.skip("Secrets test only applies on Juju 3.x")
-
-
-@pytest.fixture
-def only_without_juju_secrets(juju_has_secrets):
-    """Pretty way to skip Juju 2-specific tests.
-
-    Typically: to save CI time, when the same check were executed in a Juju 3-specific way already
-    """
-    if juju_has_secrets:
-        pytest.skip("Skipping legacy secrets tests")
```

Comment:
We had already the whole test suite running with and without secrets (`unit-juju2` and `unit-juju3` respectively), the only change needed was to adapt the few tests that only ran in one or the other, and remove this fixtures.

Reply:
merging those 2 suites into one will be addressed in a follow-up PR!